Controllers/schoolController.js

```diff
@@ -54,4 +54,59 @@ exports.getAllSchool = async (req,res)=>{
         message:error.message
        }) 
     }
+};
+
+exports.updateSchool = async (req, res) => {
+    try{
+        const {id} = req.params;
+        const {schoolName,address,principal,moto} = req.body;
+
+        const school = await schoolModel.findById(id);
+        if(!school){
+            return res.status(404).json({
+                message: "School not found!"
+            });
+        }
+
+        if (schoolName) school.schoolName = schoolName;
+        if (address) school.address = address;
+        if (principal) school.principal = principal;
+        if (moto) school.moto = moto;
+        
+        const updatedSchool = await schoolModel.save()
+
+        res.status(200).json({
+            message: "School info updated successfully",
+            data: updatedSchool
+        });
+
+    }catch(error){
+        res.status(500).json({
+            message: error.message
+        });
+    }
+};
+
+exports.deleteSchool = async (req, res) => {
+    try{
+        const {id} = req.params;
+
+        const school = await schoolModel.findById(id);
+        if (!school){
+            res.status(404).json({
+                message: "School not found!"
+            });
+        }
+
+        await school.deleteOne();
```

Comment:
Im not sure of this query method here boss. Please review it
```suggestion
        await school.findByIdAndDelete();
```